_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -1014,6 +1014,12 @@ mfxStatus VideoDECODEVP9_HW::DecodeFrameCheck(mfxBitstream *bs, mfxFrameSurface1
     sts = CheckBitstream(bs);
     MFX_CHECK_STS(sts);
 
+    if (m_vPar.mfx.FrameInfo.FourCC != surface_work->Info.FourCC)
```

Comment:
Maybe it would be  better to check this after _UpdateVideoParam_?

Reply:
yes